discover/portal/articles/210-setting-up/03-script-console/02-running-scripts-from-script-console.markdown

```diff
@@ -0,0 +1,85 @@
+# Running Scripts From the Script Console [](id=running-scripts-from-the-script-console)
+
+The script console provides a single view for executing Groovy scripts on @product@ and printing their output. The console has predefined variables that facilitate printing output and working with portlets and @product@ users. Here you'll learn these things:
```

Comment:
I can't recall, but I don't think we can call services from OSGI thru the scripting console?  Looks like the we have validated we can access OSGi services from Workflow.  Can you validate that we can call OSGi services from Scripting Console?

@alee8888 do we have any test plans for scripting console + scripting engine?  If not please add it to the list test plans to back fill.

Reply:
@mhan810 we do not have any test plans for scripting console + scripting engine. Am adding to the list of test plans missing coverage (https://docs.google.com/document/d/1XgRDlXsNg-zIpFHqtYUyBgoqFXbm-SgaGhqu1TjVXDw/edit?usp=sharing)

Reply:
@mhan810 - Yep, I was able to call services from OSGi through the scripting console. Here's a script I used …

    import com.liferay.journal.model.JournalArticle;
    import com.liferay.journal.service.JournalArticleLocalService;
    import com.liferay.portal.scripting.groovy.internal.GroovyExecutor;

    import org.osgi.framework.Bundle;
    import org.osgi.framework.FrameworkUtil;
    import org.osgi.util.tracker.ServiceTracker;

    ServiceTracker<JournalArticleLocalService, JournalArticleLocalService> st;

    try {
        Bundle bundle = FrameworkUtil.getBundle(GroovyExecutor.class);

        st = new ServiceTracker(bundle.getBundleContext(), JournalArticleLocalService.class, null);
        st.open();

        JournalArticleLocalService jaService = st.waitForService(500);
        
        if (jaService == null) {
            out.println("The required service 'JournalArticleLocalService' is not available.");
        }
        else {
            java.util.List<JournalArticle>articles = jaService.getArticles();
            if (articles != null) {
                out.println("Article count: " + articles.size());
            } else {
                out.println("no articles");
            }
        }
    }
    catch(Exception e) {
        //Handle error appropriately
    }
    finally {
        if (st != null) {
            st.close();
        }
    }